src/Main.cpp

```diff
@@ -77,30 +80,50 @@ static void httpStopPutHandler(const HttpRequest&, HttpResponse& resp) {
 }
 
 int main(int argc, char** argv) {
-    if(argc < 2) {
-        std::cerr << "Usage: " << argv[0] << " <port>" << std::endl;
-        return 1;
+
+    int c;
+    int httpPort = 8080;
+    int pgsqlPort = 5432;
+
+    while ((c = getopt (argc, argv, "p:h:")) != -1)
+    switch (c) {
+        case 'h':
+        httpPort = atoi(optarg);
+        break;
+        case 'p':
+        pgsqlPort = atoi(optarg);
+        break;
+    }
+
+
+    if(httpPort <= 0 || httpPort >= 65536) {
+        std::cerr << "Invalid httpPort number!" << std::endl;
+        return 2;
     }
 
-    int port = atoi(argv[1]);
-    if(port <= 0 || port >= 65536) {
-        std::cerr << "Invalid port number!" << std::endl;
+    if(pgsqlPort <= 0 || pgsqlPort >= 65536) {
+        std::cerr << "Invalid pgsqlPort number!" << std::endl;
         return 2;
     }
 
-    HttpServer server;
-    server.addHandler(HTTP_GET, "(?i)^/$", &httpIndexGetHandler);
-    server.addHandler(HTTP_PUT, "(?i)^/v1/_stop/?$", &httpStopPutHandler);
-    server.addHandler(HTTP_GET, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVGetHandler);
-    server.addHandler(HTTP_GET, "(?i)^/v1/kv/([\\w-]+)/([\\w-]+)/?$", &httpKVGetRangeHandler);
-    server.addHandler(HTTP_PUT, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVPutHandler);
-    server.addHandler(HTTP_DELETE, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVDeleteHandler);
+    HttpServer httpServer;
+
+    PgsqlServer pgsqlServer(&terminate_flag);
 
-    server.listen("127.0.0.1", port);
+    httpServer.addHandler(HTTP_GET, "(?i)^/$", &httpIndexGetHandler);
+    httpServer.addHandler(HTTP_PUT, "(?i)^/v1/_stop/?$", &httpStopPutHandler);
+    httpServer.addHandler(HTTP_GET, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVGetHandler);
+    httpServer.addHandler(HTTP_GET, "(?i)^/v1/kv/([\\w-]+)/([\\w-]+)/?$", &httpKVGetRangeHandler);
+    httpServer.addHandler(HTTP_PUT, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVPutHandler);
+    httpServer.addHandler(HTTP_DELETE, "(?i)^/v1/kv/([\\w-]+)/?$", &httpKVDeleteHandler);
+
+    pgsqlServer.listen("127.0.0.1", pgsqlPort);
+    httpServer.listen("127.0.0.1", httpPort);
 
     while(!terminate_flag.load()) {
-        server.accept(terminate_flag);
+        pgsqlServer.accept(terminate_flag);
+        httpServer.accept(terminate_flag);
```

Comment:
I believe it's worth adding a comment that unlike POSIX accept() procedure none of these methods is blocking.